internal/db/db.go

```diff
@@ -56,6 +60,14 @@ func (db *DB) Open(ext extensions.Extensions, bootstrap bool, project string) er
 
 	err = cluster.PrepareStmts(db.db, project, false)
 	if err != nil {
+		// If we receive an error at this point, close the database.
+		closeErr := db.db.Close()
+		if closeErr != nil {
```

Comment:
Should we also do this as part of the reverter?

Reply:
No, because we expect "skippable" errors from `waitUpgrade`. If that returns an error, we don't want to close the database because we want to retry the upgrade without re-opening the database. If we put it in the reverter, then the database will always close.

Reply:
Ok but why not putting the `reverter.Add()` right before calling `PrepareStmts`?

If we ever add new funcs after `PrepareStmts` that might return errors too we won't close the DB in case they error out. Or is closing the DB specific for if `PrepareStmts` returns an error?

Reply:
Oh I see what you mean, good point.

Reply:
Fixed.